cpp/src/arrow/compute/api_vector.h

```diff
@@ -586,6 +586,20 @@ Result<Datum> DictionaryEncode(
     const DictionaryEncodeOptions& options = DictionaryEncodeOptions::Defaults(),
     ExecContext* ctx = NULLPTR);
 
+/// \brief Compact a dictionary array
+///
+/// The output removes unused values in dictionary from the input.
+/// The function assumes every indice is effective.
```

Comment:
Could you reword this to be more explicit about what is assumed about each index? I'm not sure what you mean by "effective"

Reply:
Sure, I change to the following words:
The function assumes every indice has its corresponding value  in dictionary.